Csv/src/Akka.Streams.Csv.Tests/dsl/CsvParsingSpec.cs

```diff
@@ -0,0 +1,247 @@
+        }
+
+        [Fact]
+        public void CsvParsing_should_parse_semicolon_lines()
+        {
+            var fut = Source
+                .Single(ByteString.FromString("eins;zwei;drei\nein”s;zw ei;dr\\ei\nun’o;dos;tres\n"))
+                .Via(CsvParsing.LineScanner(delimiter: CsvParsing.SemiColon, escapeChar: 0x2a/*'*'*/))
+                .Select(list =>
+                {
+                    var outList = new List<string>();
+                    foreach (var bs in list)
+                    {
+                        outList.Add(bs.DecodeString());
+                    }
+                    return outList.ToArray();
+                })
+                .RunWith(Sink.Seq<string[]>(), Materializer);
+
+            fut.Wait(TimeSpan.FromSeconds(3));
+            var res = fut.Result;
+            res[0].ShouldAllBeEquivalentTo(new[] { "eins", "zwei", "drei" });
+            res[1].ShouldAllBeEquivalentTo(new[] { "ein”s", "zw ei", "dr\\ei" });
+        }
+
+        [Fact]
+        public void CsvParsing_should_parse_chunks_successfully()
+        {
+            var input = new[]
+            {
+                ByteString.FromString("eins,zw"),
+                ByteString.FromString("ei,drei\nuno"),
+                ByteString.FromString(",dos,tres\n")
+            };
+            var fut = Source
+                .From(input)
+                .Via(CsvParsing.LineScanner())
+                .Select(list =>
+                {
+                    var outList = new List<string>();
+                    foreach (var bs in list)
+                    {
+                        outList.Add(bs.DecodeString());
+                    }
+                    return outList.ToArray();
+                })
+                .RunWith(Sink.Seq<string[]>(), Materializer);
+            fut.Wait(TimeSpan.FromSeconds(3));
+            var res = fut.Result;
+            res[0].ShouldAllBeEquivalentTo(new[] { "eins", "zwei", "drei" });
+            res[1].ShouldAllBeEquivalentTo(new[] { "uno", "dos", "tres" });
+        }
+
+        [Fact]
+        public void CsvParsing_should_emit_completion_even_without_new_line_at_end()
+        {
+            var t = this.SourceProbe<ByteString>()
+                .Via(CsvParsing.LineScanner())
+                .Select(list =>
+                {
+                    var outList = new List<string>();
+                    foreach (var bs in list)
+                    {
+                        outList.Add(bs.DecodeString(Encoding.UTF8));
+                    }
+                    return outList.ToArray();
+                })
+                .ToMaterialized(this.SinkProbe<string[]>(), Keep.Both)
+                .Run(Materializer);
+            var source = t.Item1;
+            var sink = t.Item2;
+
+            source.SendNext(ByteString.FromString("eins,zwei,drei\nuno,dos,tres\n1,2,3"));
+            sink.Request(3);
+            sink.ExpectNext().ShouldAllBeEquivalentTo(new[] { "eins", "zwei", "drei" });
```

Comment:
`sink.ExpectNext(new[] { "eins", "zwei", "drei" })` or if it would have been three single messages you could use `sink.ExpectNext("eins", "zwei", "drei")`